policykit/policyengine/engine.py

```diff
@@ -0,0 +1,309 @@
+import logging
+
+from actstream import action as actstream_action
+from django.conf import settings
+
+from policyengine.utils import ActionKind
+
+logger = logging.getLogger(__name__)
+db_logger = logging.getLogger("db")
+
+
+class EvaluationLogAdapter(logging.LoggerAdapter):
+    def process(self, msg, kwargs):
+        kwargs["extra"] = self.extra
+        return (msg, kwargs)
+
+
+class EvaluationContext:
+    """
+    Class to hold all variables available in a policy evaluation.
+    All attributes on this class are in scope and can be used by the policy author.
+
+    Attributes:
+        proposal (Proposal): The proposal representing this evaluation.
+        action (BaseAction): The action that triggered this policy evaluation.
+        policy (Policy): The policy being evaluated.
+        slack (SlackCommunity)
+        discord (DiscordCommunity)
+        discourse (DiscourseCommunity)
+        reddit (RedditCommunity)
+        metagov (Metagov): Metagov library for performing enabled actions and processes.
```

Comment:
This is staying the same, for now. I'll do a follow up PR to change this, so we have light-weight integrations (for Loomio, Open Collective, SourceCred) instead of a generic "metagov"  class. I'm still not 100% sure what that API will look like for starting voting processes and performing actions, but that's the next think I'll tackle. @amyxzhang